From: Ralf Baechle <ralf@linux-mips.org>
Date: Tue, 6 Jan 2009 23:07:20 +0000 (+0000)
Subject: MIPS: Only write c0_framemask on CPUs which have this register.
X-Git-Url: http://www.linux-mips.org/git?p=linux.git;a=commitdiff_plain;h=75daa304755386ae7280d1738abd9df662bc97d3

MIPS: Only write c0_framemask on CPUs which have this register.

Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
(cherry picked from commit 9ad45359a5860e978a07845786148a4fdbce6f8d)
---

diff --git a/arch/mips/mm/tlb-r4k.c b/arch/mips/mm/tlb-r4k.c
index 5ce2fa7..9619f66 100644
--- a/arch/mips/mm/tlb-r4k.c
+++ b/arch/mips/mm/tlb-r4k.c
@@ -478,7 +478,10 @@ void __cpuinit tlb_init(void)
 	probe_tlb(config);
 	write_c0_pagemask(PM_DEFAULT_MASK);
 	write_c0_wired(0);
-	write_c0_framemask(0);
+	if (current_cpu_type() == CPU_R10000 ||
+	    current_cpu_type() == CPU_R12000 ||
+	    current_cpu_type() == CPU_R14000)
+		write_c0_framemask(0);
 	temp_tlb_entry = current_cpu_data.tlbsize - 1;
 
         /* From this point on the ARC firmware is dead.  */
